app/src/test/java/ch/epfl/sweng/sweng_golf/OfferTest.java

```diff
@@ -0,0 +1,23 @@
+package ch.epfl.sweng.sweng_golf;
+
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+
+public class OfferTest {
+    @Test(expected = IllegalArgumentException.class)
+    public void testInvalidParameters(){
+        Offer offer = new Offer("","","");
+
+    }
+
```

Comment:
Maybe trying to test each case (username valid but not title, username & title but not description) could help improve the code coverage?

Reply:
I added some tests cover these cases.